pallets/foreign-investments/src/impls/mod.rs

```diff
@@ -56,6 +54,11 @@ impl<T: Config> ForeignInvestment<T::AccountId> for Pallet<T> {
 			!T::Investment::investment_requires_collect(who, investment_id),
 			Error::<T>::InvestError(InvestError::CollectRequired)
 		);
+
+		// NOTE: For the MVP, we simply overwrite any existing payment currency. In a
+		// later version, we might want to store a `BoundedVec` instead.
```

Comment:
I almost always find it better to use a double map instead with `()` as the value type since that makes storage reads/writes a lot cheaper and keeps the code simple as it doesn't need the Max for the bounded vec. If this is something we may need in the future, it's low effort to implement it already and save the migration work later on.

Reply:
I think both options have separate use cases. When using a `DoubleStorageMap`, you don't want to limit the key-value pairs which you want when using a `StorageMap` with `BoundedVec` values. 

Nevertheless, I agree a double map might be more suitable here. Given that we currently limit `(investor, investment_id)` pairs to have at most one swap at the same time, adding true multicurrency support (both in EVM as well as Centrifuge Chain) will be a bigger feature. It was discussed this is something we need mid- to long-term (rather 2024) but not for this MVP.

Reply:
> [It was discussed ](https://centrifuge.hackmd.io/tKGS5CwqSQeeI3bU1dKUlw#20230815)this is something we need mid- to long-term (rather 2024) but not for this MVP.


Then I say we do this now, it takes very little effort compared to the significant effort and error-proneness that writing a migration for this in the future will entail.

Reply:
I see it's done already, thanks for doing that :)

Reply:
I just updated the comment but did not alter any storage. Migrating to double maps will touch multiple storages and complicates swap order handling much more. It definitely should not be part of this PR.

Reply:
Oh, I looked at the `InvestmentPaymentCurrency` which is a double map, my bad. I don't agree with the decision here but it's not a blocker for me.